src/Contracts/OpdexMiningGovernance.cs

```diff
@@ -170,15 +163,19 @@ public void RewardMiningPools()
 
         var nominations = Nominations;
         
-        for (uint i = 0; i < MaxNominations; i++)
+        for (uint i = 0; i < nominations.Length; i++)
         {
-            if (nominations[i].StakingPool == Address.Zero) continue;
+            if (nominations[i].Weight == UInt256.Zero) continue;
             
             RewardMiningPoolExecute(nominations[i], MiningPoolReward);
             IncrementMiningPoolsFunded();
+            
+            nominations[i].Weight = UInt256.Zero;
```

Comment:
Still problematic, if nobody stakes and the previous 4 liquidity pools are used again, they would've had their weight reset to 0, bypassing rewarding the mining pools.